### PR TITLE
Add tool search, description, and schema query functionalities 

### PR DESCRIPTION
**Pull request by gauravlochab**
This PR adds new features enabling users to search for tools using agent-id, and to fetch detailed descriptions and input/output schemas for these tools using their unique identifier. 
Additionally, these functionalities have been integrated into the CLI, and detailed usage instructions have been included in the README file.

### CODE REVIEW

File: mech_client/cli.py [hunk 1]
```diff
@@ -1,31 +1,18 @@
-# -*- coding: utf-8 -*-
```
Comment (DavidMinarsch) [inline mech_client/cli.py:1]: These need to stay

Comment (gauravlochab) [reply mech_client/cli.py]: added back license header

File: README.md
Comment (0xArdi) [inline README.md:1]: Nice explanation!

File: mech_client/mech_tool_management.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    )
+    response_json = response.json()
+
+    # Try to extract the ABI from different possible response structures
+    if "result" in response_json and isinstance(response_json["result"], str):
+        contract_abi = json.loads(response_json["result"])
+    elif "abi" in response_json:
+        contract_abi = response_json["abi"]
+    else:
+        raise ValueError(f"Unexpected API response structure: {response_json}")
```
[the remark this answers is not in the document]
Comment (gauravlochab) [reply mech_client/mech_tool_management.py]: pushed these changes

File: pyproject.toml [hunk 3]
```diff
@@ -33,6 +33,7 @@ open-aea-ledger-cosmos = "^1.53.0"
 gql = ">=3.4.1"
 asn1crypto = ">=1.4.0,<1.5.0"
 websocket-client = ">=0.32.0,<1"
+tabulate = "^0.9.0"
```
[the remark this answers is not in the document]
Comment (gauravlochab) [reply pyproject.toml:36]: tabulate is used to display data for tool input/output schemas and lists of tools available for agents or specific agent  into structured tables that makes the response easier to read and understand.

File: mech_client/mech_tool_management.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+    # Create a contract instance
+    contract = w3.eth.contract(
+        address=mech_config.agent_registry_contract, abi=contract_abi
+    )
```
[the remark this answers is not in the document]
Comment (gauravlochab) [reply mech_client/mech_tool_management.py]: pushed the changes

**Review by 0xArdi — APPROVED** [review]

**Review by Adamantios — APPROVED** [review]

**Pull request merged**